telemetry/sql/000015_common_fields.up.sql

```diff
@@ -0,0 +1,118 @@
+CREATE TABLE IF NOT EXISTS commonFields (
+    id SERIAL PRIMARY KEY,
+    nodeName VARCHAR(255) NOT NULL,
+    peerId VARCHAR(255) NOT NULL,
+    statusVersion VARCHAR(255) NOT NULL,
+    deviceType VARCHAR(255) NOT NULL,
+    createdAt INTEGER DEFAULT EXTRACT(EPOCH FROM NOW())::INTEGER
+);
+
+-- Function to modify tables with added logging
+CREATE OR REPLACE FUNCTION modify_table(table_name TEXT) RETURNS VOID AS $$
+BEGIN
+
+    -- Add the new column for the foreign key
+    EXECUTE format('ALTER TABLE %I ADD COLUMN IF NOT EXISTS commonFieldsId INTEGER', table_name);
```

Comment:
Not pushing on termins here, just sharing opniong.

I would call this column just `id`. And the `commonFields`, would simply be `TelemetryRecord`. And then there're difference extensions of these records that reference the main record by ID.

```suggestion
    EXECUTE format('ALTER TABLE %I ADD COLUMN IF NOT EXISTS id INTEGER', table_name);
```

Reply:
Do you mean multiple rows in other tables sharing the same `TelemetryRecord`? e.g. a row in `receivedMessages` and a row in `peercount` with the same value in column `id`?

I think this would work but would require removing the timestamp from `TelemetryRecord` and putting it back into the other tables.

Reply:
Nah, never mind. After second thought on this, I decided to google how to do it right 😄 
What we're doing here is called [Class Table Inheritance](https://stackoverflow.com/a/3579462/7449458).

Reply:
So my suggestion would be just to rename `CommonFields` to `TelemetryRecord` (or simply `Record`).
And consequently rename `commonFieldsId` -> `record_id`. 

But this is not just my subjective opinion, leaving this at your decision ofc.

Reply:
Renamed in latest commit